pkg/virtualization/core/storage/service/ImageManagementService.go

```diff
@@ -173,3 +173,57 @@ func (ims *ImageManagementService) CreateDisk(settings *disk.VirtualHardDiskSett
 	defer job.Close()
 	return job.WaitForJobCompletion(result.ReturnValue, -1)
 }
+
+func (ims *ImageManagementService) GetVirtualHardDiskConfig(path string) (size uint64,
+	blockSize uint32,
+	lSectorSize uint32,
+	pSectorSize uint32,
+	format uint16,
+	err error) {
+	method, err := ims.GetWmiMethod("GetVirtualHardDiskSettingData")
+	if err != nil {
+		return
+	}
+	defer method.Close()
+
+	inparams := wmi.WmiMethodParamCollection{}
+	inparams = append(inparams, wmi.NewWmiMethodParam("Path", path))
+
+	outparams := wmi.WmiMethodParamCollection{}
+	outparams = append(outparams, wmi.NewWmiMethodParam("SettingData", nil))
+	outparams = append(outparams, wmi.NewWmiMethodParam("Job", nil))
+
+	result, err := method.Execute(inparams, outparams)
+	if err != nil {
+		return
+	}
+
+	if !(result.ReturnValue == 4096 || result.ReturnValue == 0) {
```

Comment:
Would you please add some comments to explain the return values (Job started == 4096 and Completed with no error == 0)? Otherwise, this is hard to understand what it does.

Reply:
These return codes are used all across the WMI repository and changing it just at this place would be an incomplete task. Let me take this comment and update all the places in a separate PR.